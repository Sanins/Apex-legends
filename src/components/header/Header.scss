.header {
    display: grid;
    min-height: 100px;
    grid-auto-flow: column;
    grid-gap: 1.5em;
    padding: 0 40px;
    box-shadow: -1px 4px 5px 0px rgba(0,0,0,1);

    &__logo {
        color: #FFF;
        display: grid;
        justify-content: start;
        align-content: center;
    }

    &__links {
        display: grid;
        justify-content: right;
        align-content: center;
        grid-auto-flow: column;
        grid-gap: 1.5em;
    }
}