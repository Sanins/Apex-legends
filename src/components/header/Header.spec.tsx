import { shallow } from 'enzyme';
import * as React from 'react';
import Header from './Header';
import { useHeaderElements } from './Header'
import { BrowserRouter } from 'react-router-dom';

describe('<Header/>', () => {
    const container = shallow(<BrowserRouter><Header /></BrowserRouter>);
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});

describe('useHeaderElements', () => {
    const Elements = () => {
      const props = useHeaderElements({});
      return <div {...props} />;
    };
    const container = shallow(<Elements />);
  
    it('should have proper props for email field', () => {
      expect(container.prop('btnClick')).toEqual({
        toggle: expect.any(Function),
        close: expect.any(Function),
        currentValue: false
      });
    });
  });