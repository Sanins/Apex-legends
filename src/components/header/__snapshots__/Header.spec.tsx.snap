// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match the snapshot 1`] = `"<div class=\\"logo\\">my logo</div><button class=\\"header__mobile-hamburger\\">Show/hide</button><ul class=\\"header header__close\\"><a aria-current=\\"page\\" class=\\"header__list-item header__list-item--selected\\" href=\\"/\\"><div class=\\"header__list-item__link\\">Home</div></a><a class=\\"header__list-item\\" href=\\"/legends\\"><div class=\\"header__list-item__link\\">Legends</div></a><a class=\\"header__list-item\\" href=\\"/weapons\\"><div class=\\"header__list-item__link\\">Weapons</div></a><a class=\\"header__list-item\\" href=\\"/items\\"><div class=\\"header__list-item__link\\">items</div></a></ul>"`;

exports[`<NewHeader/> should match the snapshot 1`] = `"<div class=\\"logo\\">my logo</div><button class=\\"header__mobile-hamburger\\">Show/hide</button><ul class=\\"header header__close\\"><a aria-current=\\"page\\" class=\\"header__list-item header__list-item--selected\\" href=\\"/\\"><div class=\\"header__list-item__link\\">Home</div></a><a class=\\"header__list-item\\" href=\\"/legends\\"><div class=\\"header__list-item__link\\">Legends</div></a><a class=\\"header__list-item\\" href=\\"/weapons\\"><div class=\\"header__list-item__link\\">Weapons</div></a><a class=\\"header__list-item\\" href=\\"/items\\"><div class=\\"header__list-item__link\\">items</div></a></ul>"`;
